models/vit_gpt2_image_captioning.py

```diff
@@ -0,0 +1,108 @@
+import datetime
+import json
+import os
+import shutil
+import dtlpy as dl
+import torch
+from transformers import VisionEncoderDecoderModel, ViTImageProcessor, AutoTokenizer
+import torch
+from PIL import Image
+
+
+def create_folder(folder):
+    if os.path.exists(folder):
+        shutil.rmtree(folder, ignore_errors=True)
+    os.mkdir(folder)
+
+
+class HuggingAdapter:
+    def __init__(self, configuration):
+        self.model_name = configuration.get("model_name")
+        model_id = "nlpconnect/vit-gpt2-image-captioning"
+        self.model = VisionEncoderDecoderModel.from_pretrained(model_id)
+        self.feature_extractor = ViTImageProcessor.from_pretrained(model_id)
+        self.tokenizer = AutoTokenizer.from_pretrained(model_id)
+        self.device = "cuda"
+        self.model.to(self.device)
+        self.gen_kwargs = {"max_length": configuration.get("max_length", 16),
+                           "num_beams": configuration.get("num_beams", 4)}
+        self.results_local_path = "vit_gpt2_image_captioning_results"
+        create_folder(self.results_local_path)
+
+    def prepare_item_func(self, item: dl.Item):
+        buffer = json.load(item.download(save_locally=False))
+        prompts = buffer["prompts"]
+        ready_prompts = []
+        for prompt_key, prompt_content in prompts.items():
+            if "image" in prompt_content[0]["mimetype"]:
```

Comment:
It's a similar case here. `prompt_content` should not be a list, but a dictionary. We should use the first value, not just the index 0 of a list.